app/src/main/cpp/TrackedKeyboardRenderer.cpp

```diff
@@ -0,0 +1,440 @@
+        if (result != KTX_SUCCESS) {
+            VRB_LOG("XR_FB_keyboard_tracking: Error transcoding KTX texture: (%u) %s", result,
+                    ktxErrorString(result));
+            ktxTexture_Destroy(*ktxTex);
+            *ktxTex = nullptr;
+            return false;
+        }
+    }
+
+    VRB_LOG("XR_FB_keyboard_tracking: KTX texture created successfully");
+
+    return true;
+}
+
+bool TrackedKeyboardRenderer::LoadKeyboardMesh(const std::vector<uint8_t>& modelBuffer) {
+    VRB_LOG("XR_FB_keyboard_tracking: Load keyboard mesh");
+
+    // Destroy any previous model
+    if (m.model != nullptr) {
+        m.model.reset();
+        m.model = nullptr;
+    }
+
+    GLTFModel gltfModel;
+    tinygltf::TinyGLTF modelLoader;
+    auto& model = gltfModel.gltf;
+    std::string err;
+    std::string warn;
+    ktxTexture* ktxTex = nullptr;
+
+    // Use libktx (from KTX-Software suite) to load textures in the glTF file
+    modelLoader.SetImageLoader(&TrackedKeyboardRenderer::ImageLoaderCallback, &ktxTex);
+
+    if (!modelLoader.LoadBinaryFromMemory(&model, &err, &warn, modelBuffer.data(), modelBuffer.size())) {
+        VRB_ERROR("XR_FB_keyboard_tracking: Error loading keyboard model: %s", err.c_str());
+        return false;
+    }
+
+    if (!warn.empty())
+        VRB_WARN("%s", warn.c_str());
+
+    if (model.meshes.size() == 0)
+        return false;
+    // Assume the first mesh
+    auto& mesh = model.meshes[0];
+
+    if (mesh.primitives.size() == 0)
+        return false;
+    // Assume the first primitive of the mesh
+    auto& primitive = mesh.primitives[0];
+
+    // Load indices
+    if (primitive.indices < 0 || primitive.indices >= model.accessors.size())
+        return false;
+    gltfModel.indicesAttrIndex = primitive.indices;
+
+    // Load vertex attributes
+    for (auto& attr: primitive.attributes) {
+        if (attr.second >= model.accessors.size())
+            return false;
+
+        if (attr.first == "POSITION") {
+            gltfModel.positionAttrIndex = attr.second;
+        } else if (attr.first == "NORMAL") {
+            gltfModel.normalAttrIndex = attr.second;
+        } else if (attr.first == "TEXCOORD_0") {
+            gltfModel.texcoordAttrIndex = attr.second;
+        }
+    }
+
+    if (gltfModel.positionAttrIndex == -1 || gltfModel.normalAttrIndex == -1 || gltfModel.texcoordAttrIndex == -1)
+        return false;
+
+    m.model = std::make_unique<GLTFModel>(gltfModel);
+    m.model->ktxTex = ktxTex;
+
+    VRB_LOG("XR_FB_keyboard_tracking: Keyboard model loaded successfully");
+
+    return true;
+}
+
+void TrackedKeyboardRenderer::SetVisible(const bool aVisible) {
+    m.visible = aVisible;
+}
+
+bool TrackedKeyboardRenderer::InitializeGL() {
+    m.vertexShader = vrb::LoadShader(GL_VERTEX_SHADER, sVertexShader);
+    m.fragmentShader = vrb::LoadShader(GL_FRAGMENT_SHADER, sFragmentShader);
+    if (m.vertexShader && m.fragmentShader)
```

Comment:
As we are the ones creating the shaders we should be confident enough that they will be properly loaded and thus I think this should be an assert instead. Or is there a risk in your opinion that due to the multiple different architectures we support the shaders could not be parsed correctly?